.github/actions/compile-commit/action.yml

```diff
@@ -12,22 +12,22 @@ runs:
       shell: bash
       run: |
         set -x
-        
+
         # The section below should always contain copies of env variables from .github/workflows/ci.yml to maintain both files in sync
         # It's important that these values are NOT passed as parameters, because then their values would always be taken from PR HEAD
         # -------
         # allow overriding Maven command
-        export MAVEN="./mvnw"
+        MAVEN="./mvnw"
         # maven.wagon.rto is in millis, defaults to 30m
-        export MAVEN_OPTS="-Xmx512M -XX:+ExitOnOutOfMemoryError -Dmaven.wagon.rto=60000"
-        export MAVEN_INSTALL_OPTS="-Xmx3G -XX:+ExitOnOutOfMemoryError -Dmaven.wagon.rto=60000"
-        export MAVEN_COMPILE_COMMITS="-B --strict-checksums --quiet -T C1 -DskipTests -Dmaven.source.skip=true -Dair.check.skip-all=true -Dmaven.javadoc.skip=true --no-snapshot-updates --no-transfer-progress -pl !:trino-server-rpm"
-        export MAVEN_GIB="-P gib -Dgib.referenceBranch=refs/remotes/origin/${{ inputs.base_ref }}"
+        MAVEN_INSTALL_OPTS="-Xmx3G -XX:+ExitOnOutOfMemoryError -Dmaven.wagon.rto=60000"
+        MAVEN_COMPILE_COMMITS="-B --strict-checksums --quiet -T C1 -DskipTests -Dmaven.source.skip=true -Dair.check.skip-all=true -Dmaven.javadoc.skip=true --no-snapshot-updates --no-transfer-progress -pl !:trino-server-rpm"
+        MAVEN_GIB="-P gib -Dgib.referenceBranch=refs/remotes/origin/${{ inputs.base_ref }}"
+        RETRY: .github/bin/retry
         # -------
-        
+
         # For building with Maven we need MAVEN_OPTS to equal MAVEN_INSTALL_OPTS
         export MAVEN_OPTS="${MAVEN_INSTALL_OPTS}"
-        $MAVEN package ${MAVEN_COMPILE_COMMITS} ${MAVEN_GIB}
+        $RETRY $MAVEN package ${MAVEN_COMPILE_COMMITS} ${MAVEN_GIB}
```

Comment:
If compilation fails, it'll be retrying it until the whole job times out. Instead of adding retries here, maybe it's better to add a step with `$RETRY $MAVEN dependency:go-offline`, similar to what we already do in the `maven-checks` job.

Reply:
Thanks, will do.

Do you want to apply this to `$RETRY $MAVEN clean install ${MAVEN_FAST_INSTALL} ${MAVEN_GIB} -am -pl "${{ matrix.modules }}"` in the ci's main `test` job?

Reply:
> Do you want to apply this to `$RETRY $MAVEN clean install ${MAVEN_FAST_INSTALL} ${MAVEN_GIB} -am -pl "${{ matrix.modules }}"` in the ci's main `test` job?

@nineinchnick ?